pkg/diagnostics/tracing.go

```diff
@@ -77,13 +78,20 @@ func TraceSpanFromFastHTTPRequest(r *fasthttp.Request, spec config.TracingSpec)
 
 	corID := string(r.Header.Peek(CorrelationID))
 	uriSpanName := string(r.Header.RequestURI())
+	rate := diag_utils.GetTraceSamplingRate(spec.SamplingRate)
+
+	// TODO : Continue using ProbabilitySampler till Go SDK starts supporting RateLimiting sampler
+	probSamplerOption := trace.WithSampler(trace.ProbabilitySampler(rate))
+	serverKindOption := trace.WithSpanKind(trace.SpanKindServer)
+	clientKindOption := trace.WithSpanKind(trace.SpanKindClient)
+	spanName := createSpanName(uriSpanName)
 	if corID != "" {
 		spanContext := DeserializeSpanContext(corID)
-		ctx, span = trace.StartSpanWithRemoteParent(context.Background(), uriSpanName, spanContext, trace.WithSpanKind(trace.SpanKindServer))
-		ctxc, spanc = trace.StartSpanWithRemoteParent(ctx, createSpanName(uriSpanName), span.SpanContext(), trace.WithSpanKind(trace.SpanKindClient))
+		ctx, span = trace.StartSpanWithRemoteParent(context.Background(), uriSpanName, spanContext, serverKindOption, probSamplerOption)
+		ctxc, spanc = trace.StartSpanWithRemoteParent(ctx, spanName, span.SpanContext(), clientKindOption, probSamplerOption)
 	} else {
-		ctx, span = trace.StartSpan(context.Background(), uriSpanName, trace.WithSpanKind(trace.SpanKindServer))
-		ctxc, spanc = trace.StartSpanWithRemoteParent(ctx, createSpanName(uriSpanName), span.SpanContext(), trace.WithSpanKind(trace.SpanKindClient))
+		ctx, span = trace.StartSpan(context.Background(), uriSpanName, serverKindOption, probSamplerOption)
+		ctxc, spanc = trace.StartSpanWithRemoteParent(ctx, spanName, span.SpanContext(), clientKindOption, probSamplerOption)
```

Comment:
This block seems too repetitive, can be moved to a method

Reply:
Refactored to getTraceSpan method for http requests. However leaving out gRPC one as it is only one method that creating tracespan, as parent /child span tight, so keeping that one intact.